src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstring.cs

```diff
@@ -0,0 +1,135 @@
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager,
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        private static readonly LocalizableString s_localizableTitle2 = new LocalizableResourceString(
+            nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterTitle),
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, 
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessageDefault2 = new LocalizableResourceString(
+            nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterDefault),
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager,
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription2 = new LocalizableResourceString(
+            nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterDescription),
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, 
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor RuleReplaceOneParameter = new DiagnosticDescriptor(
+            RuleIdOneParameterId,
+            s_localizableTitle,
+            s_localizableMessageDefault,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: s_localizableDescription,
+            helpLinkUri: "", // TODO (might require an addition to docs repository?
+            customTags: WellKnownDiagnosticTags.AnalyzerException);
+
+        internal static DiagnosticDescriptor RuleReplaceTwoParameter = new DiagnosticDescriptor(
+            RuleIdTwoParameterId,
+            s_localizableTitle2,
+            s_localizableMessageDefault2,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: s_localizableDescription2,
+            helpLinkUri: "", // TODO (might require an addition to docs repository?
+            customTags: WellKnownDiagnosticTags.AnalyzerException);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleReplaceOneParameter, RuleReplaceTwoParameter);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(this.Action, SyntaxKind.InvocationExpression);
```

Comment:
Please switch this to be an `IOperation` based analyzer. This will likely enable you to remove references to C# syntax nodes/kinds. You should be able to switch this to a language-agnostic analyzer that works for both C# and VB. Please look at other operation analyzers in this repo (there are plenty of them, especially ones registering `OperationKind.Invocation`).

Reply:
thanks for the input, missed the right APIs, yet.
Refactored now and Analyzer is working (see upcoming commit).